lib/ansible/module_utils/facts/network/linux.py

```diff
@@ -147,9 +147,13 @@ def get_interfaces_info(self, ip_path, default_ipv4, default_ipv6):
                 interfaces[device]['perm_macaddress'] = get_file_content(os.path.join(path, 'bonding_slave', 'perm_hwaddr'), default='')
             if os.path.exists(os.path.join(path, 'device')):
                 interfaces[device]['pciid'] = os.path.basename(os.readlink(os.path.join(path, 'device')))
+            if os.path.exists(os.path.join(path, 'duplex')):
+                duplex = get_file_content(os.path.join(path, 'duplex'))
+                if duplex is not None and duplex != 'unknown':
+                    interfaces[device]['duplex'] = duplex
             if os.path.exists(os.path.join(path, 'speed')):
                 speed = get_file_content(os.path.join(path, 'speed'))
-                if speed is not None:
+                if speed is not None and duplex is not None and duplex != 'unknown':
```

Comment:
I'm not sure I understand why this is dependent on the 'duplex' value? 

For /sys/class/net/* info, I'd prefer just to report what is found to try to avoid interpreting it much (we already do a little of that but I'd prefer not to add more).